engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -697,23 +711,32 @@ yield switch (identifier) {
               );
               List<IR.DefinitionArgument> args = cons(arg_, nil());
               var body = translateExpression(app.getRhs(), false);
+              if (body == null) {
+                body = new IR$Expression$Block(
+                  nil(), new IR$Name$Blank(Option.empty(), meta(), diag()),
+                  Option.empty(), true, meta(), diag()
+                );
+              }
               yield new IR$Function$Lambda(args, body, getIdentifiedLocation(tree), true, meta(), diag());
             }
           }
           default -> {
             var lhs = translateCallArgument(app.getLhs(), insideTypeSignature);
             var rhs = translateCallArgument(app.getRhs(), insideTypeSignature);
-            yield new IR$Application$Operator$Binary(
-              lhs,
-              new IR$Name$Literal(
-                op.codeRepr(), true,
-                getIdentifiedLocation(app),
-                meta(), diag()
-              ),
-              rhs,
+            var name = new IR$Name$Literal(
+              op.codeRepr(), true,
               getIdentifiedLocation(app),
               meta(), diag()
             );
+            var loc = getIdentifiedLocation(app);
+            yield lhs != null ? new IR$Application$Operator$Binary(
```

Comment:
I would generally prefer using ternary operator only when something is a oneliner